consul/src/main/scala/io/buoyant/consul/CatalogNamer.scala

```diff
@@ -83,55 +93,58 @@ class CatalogNamer(
   }
 
   /**
-   * Contains all cached serviceNodes responses for a particular serviceName in a particular datacenter
+   * Contains all cached serviceNodes responses for a particular serviceName
+   * in a particular datacenter
    */
   case class SvcCache(datacenter: String, key: SvcKey, updateableAddr: VarUp[Addr]) {
 
     def addrs: VarUp[Addr] = updateableAddr
 
-    var index = "0"
     val _ = init()
 
-    def setIndex(idx: String) = {
-      index = idx
-    }
-
-    def mkRequest(): Future[Seq[ServiceNode]] =
-      api
-        .serviceNodes(key.name, datacenter = Some(datacenter), tag = key.tag, blockingIndex = Some(index), retry = true)
-        .map { indexedNodes =>
-          indexedNodes.index.foreach(setIndex)
-          indexedNodes.value
-        }
+    def mkRequest(index: Option[String]) =
+      catalogApi.serviceNodes(
+        key.name,
+        datacenter = Some(datacenter),
+        tag = key.tag,
+        blockingIndex = index,
+        retry = true
+      )
 
     def clear(): Unit = synchronized {
       updateableAddr() = Addr.Pending
     }
 
-    def init(): Future[Unit] = mkRequest().flatMap(update).handle(handleUnexpected)
+    def init(): Future[Unit] = domainFuture.flatMap { domain =>
+      val meta = if (setHost)
+        Addr.Metadata(("authority", s"${key.name}.service.$datacenter.$domain"))
```

Comment:
Ideally, it would be great to have the known metadata keys somewhere (as we do for headers, for instance) -- we can be incremental to getting it somewhere global, though.